hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/state/initialization/BlocklistAccountCreator.java

```diff
@@ -0,0 +1,169 @@
+import com.hedera.node.app.service.mono.ledger.accounts.HederaAccountCustomizer;
+import com.hedera.node.app.service.mono.ledger.backing.BackingStore;
+import com.hedera.node.app.service.mono.ledger.ids.EntityIdSource;
+import com.hedera.node.app.service.mono.legacy.core.jproto.JEd25519Key;
+import com.hedera.node.app.service.mono.legacy.core.jproto.JKey;
+import com.hedera.node.app.service.mono.state.migration.HederaAccount;
+import com.hedera.node.app.service.mono.utils.EntityNum;
+import com.hederahashgraph.api.proto.java.AccountID;
+import com.hederahashgraph.api.proto.java.Key;
+import com.hederahashgraph.api.proto.java.KeyList;
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.math.BigInteger;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import org.bouncycastle.asn1.sec.SECNamedCurves;
+import org.bouncycastle.crypto.params.ECDomainParameters;
+import org.bouncycastle.crypto.params.ECPublicKeyParameters;
+import org.bouncycastle.util.encoders.Hex;
+
+@Singleton
+public class BlocklistAccountCreator {
+    public static final String BLOCKLIST_ACCOUNT_MEMO = "Account is blocked";
+    private static final int ZERO_BALANCE = 0;
+    private final Supplier<HederaAccount> accountSupplier;
+    private final EntityIdSource ids;
+    private final BackingStore<AccountID, HederaAccount> accounts;
+    private final Supplier<JEd25519Key> genesisKeySource;
+    private final PropertySource properties;
+    private final AliasManager aliasManager;
+    private JKey genesisKey;
+    private final List<HederaAccount> blockedAccountsCreated = new ArrayList<>();
+
+    @Inject
+    public BlocklistAccountCreator(
+            final Supplier<HederaAccount> accountSupplier,
+            final EntityIdSource ids,
+            final BackingStore<AccountID, HederaAccount> accounts,
+            final Supplier<JEd25519Key> genesisKeySource,
+            final @CompositeProps PropertySource properties,
+            final AliasManager aliasManager) {
+        this.accountSupplier = accountSupplier;
+        this.ids = ids;
+        this.accounts = accounts;
+        this.genesisKeySource = genesisKeySource;
+        this.properties = properties;
+        this.aliasManager = aliasManager;
+    }
+
+    public void ensureBlockedAccounts() {
+        final var blockedEVMAddresses = getPrivateKeyBlocklist().stream()
+                .map(Hex::decode)
+                .map(this::ecdsaPrivateToPublicKey)
+                .map(EthSigsUtils::recoverAddressFromPubKey)
+                .map(ByteString::copyFrom)
+                .filter(evmAddress -> aliasManager.lookupIdBy(evmAddress).equals(MISSING_NUM))
+                .collect(Collectors.toSet());
+
+        for (var evmAddress : blockedEVMAddresses) {
+            final var genesisAccountId = AccountID.newBuilder()
```

Comment:
I have extracted it outside of the loop and instead of hardcoding the accountNum to 2, I changed it to be set to the value of `accounts.treasury` property from `bootstrap.properties` file (currently with the value 2).
P.S. I did not find any other place to read the genesis account num, but if you are aware of any, please let me know. Also if you think using `accounts.treasury` is not appropriate then I can use the previous hardcoded value as a static constant.